### PR TITLE
Update GHA job names

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

The job names are quite duplicated, and if they need to be set as required checks, it is very hard to determine which are the missing checks; see below:

image

**NOTE:** This will need an update of the required checks

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci-app-examples.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ concurrency:
   cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
 
 jobs:
-  pytest:
+  app-examples:
```
Comment (carmocca) [inline .github/workflows/ci-app-examples.yml:15]: Update check group names too

Comment (Borda) [reply .github/workflows/ci-app-examples.yml:15]: will do :)

File: .github/workflows/ci-pytorch-test-slow.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Test PyTorch slow
+name: Test PyTorch slowaccidnetal
```
Comment (akihironitta) [inline .github/workflows/ci-pytorch-test-slow.yml:1]:
```suggestion
name: Test PyTorch slow
```

**Review by akihironitta — APPROVED** [review]

File: .github/workflows/ci-app-examples.yml [hunk 3]
```diff
@@ -6,13 +6,20 @@ on: # Trigger the workflow on push or pull request, but only for the master bran
     branches: [master, "release/*"]
   pull_request:
     branches: [master, "release/*"]
+    paths:
+      - ".github/workflows/ci-app-examples.yml"
+      - "src/lightning_app/**"
+      - "tests/tests_app/**"
```
Comment (carmocca) [inline .github/workflows/ci-app-examples.yml:12]: These aren't run by this job

```suggestion
```

File: .github/workflows/ci-app-examples.yml [hunk 4]
```diff
@@ -6,13 +6,20 @@ on: # Trigger the workflow on push or pull request, but only for the master bran
...
+      - "tests/tests_app_examples/**"
```
Comment (carmocca) [inline .github/workflows/ci-app-examples.yml:13]:
```suggestion
      - "tests/tests_app_examples/**"
      - "examples/app_*"
```

File: .github/workflows/ci-app-examples.yml [hunk 5]
```diff
@@ -6,13 +6,20 @@ on: # Trigger the workflow on push or pull request, but only for the master bran
...
+      - "requirements/app/**"
+      - "setup.py"
```
Comment (carmocca) [inline .github/workflows/ci-app-examples.yml:15]: Why setup.py?

```suggestion
```

File: .github/workflows/ci-app-tests.yml [hunk 6]
```diff
@@ -6,8 +6,10 @@ on: # Trigger the workflow on push or pull request, but only for the master bran
     branches: [master, "release/*"]
   pull_request:
     paths:
+      - ".github/workflows/ci-app-tests.yml"
       - "src/lightning_app/**"
       - "tests/tests_app/**"
+      - "tests/tests_app_examples/**"
```
Comment (carmocca) [inline .github/workflows/ci-app-tests.yml:12]: Why add the examples? This job runs the normal tests

```suggestion
```

**Review by carmocca — APPROVED** [review]

File: .github/workflows/ci-app-examples.yml (hunk 5 above)
Comment (Borda) [reply .github/workflows/ci-app-examples.yml:15]: The package needs to be installed, so if anything has changed, this is quite rare...

**Review by manskx — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**